gematria/experiments/access_pattern_bm/contiguous_matrix.cc

```diff
@@ -0,0 +1,37 @@
+#include <immintrin.h>
+
+#include <random>
+
+namespace gematria {
+
+static std::default_random_engine generator;
+static std::uniform_int_distribution<int> distribution(0, 1023);
+
+int *CreateRandomContiguousMatrix(const std::size_t size) {
+  auto matrix = new int[size * size];
+
+  for (int i = 0; i < size; ++i) {
+    for (int j = 0; j < size; ++j) {
+      matrix[size * i + j] = distribution(generator);
+    }
+  }
+
+  return matrix;
+}
+
+void FlushContiguousMatrixFromCache(int *matrix, std::size_t size) {
+  constexpr int line_size = 64;  // For all modern Intel x86 processors (?)
+  const char *ptr = (const char *)matrix;
+  const char *end = (const char *)(matrix + (size + 1) * sizeof(int));
+
+  _mm_mfence();
+  while (ptr <= end) {
+    _mm_clflushopt(ptr);
+    ptr += line_size;
+  }
+  _mm_mfence();
+}
```

Comment:
This makes sense to me, but I am not sure if it'll clean up the code in its current state given that we are only flushing multiple container elements simultaneously in `contiguous_matrix_bm` since, as far as I am aware, that is the only place where elements are guaranteed to be stored contiguously.

We can also isolate the `-mclflushopt` dependency by writing a function that flushes one container element (given that it fits on a single cache line), but that would end up being a simple wrapper around `_mm_clflushopt`, which I found to be slightly clunky. Would doing so be a better idea?

Reply:
Ack. Let's keep it the way it is.

That said, it's probably better to use `-mclflushopt` on all targets in this directlry, because it might affect which defined are present in the preprocessor, and as a consequence, make some includes (namely `<immintr.h>` look different when they are parsed for different bazel targets).